src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProjectFactory.cs

```diff
@@ -89,16 +101,19 @@ public VisualStudioProject CreateAndAddToWorkspace(string projectSystemName, str
                 if (w.CurrentSolution.ProjectIds.Count == 0)
                 {
                     // Fetch the current solution path. Since we're on the UI thread right now, we can do that.
-                    string? solutionFilePath = null;
-                    var solution = (IVsSolution)Shell.ServiceProvider.GlobalProvider.GetService(typeof(SVsSolution));
-                    if (solution != null)
+                    var solutionFilePath = _threadingContext.JoinableTaskFactory.Run(async () =>
```

Comment:
Can you add a TODO before this:

    // TODO: when ApplyChangeToWorkspace is made async, remove this JoinableTaskFactory.Run

Explanation: ApplyChangeToWorkspace takes a lock for updating the workspace, and we've also seen cases where if there's a bunch of updates happening we can end up with some thread pool threads blocked up on that. We have a tracking item somewhere to make it async and change the lock to a SempahoreSlim -- in that case then this wouldn't need the .Run() anymore since it wouldn't be sync-over-async.

Reply:
sure :)